### DISCUSSION TITLE
Restart simulation

### DISCUSSION THREAD

**Discussion by nourriahi** [idaholab/moose#25507] [Q&A General]
Hello everyone,

I am running some simulations which takes some time to end and for this I am using a remote computer that shuts down at some times, I tried to restart these simulations from the point where it breaks with the "recovery option" but this latest did not work. Is there any command that allows me to restart the simulations from the last calculated timestep?

Thanks,
Nour

**Comment by GiudGiud** [comment]
Hello

recovery using a checkpoint file is the prime way to do this.
The second option is to use a SolutionUserObject but it takes more steps
What error did you get?

Guillaume

**Comment by nourriahi** [comment]
Hello, 

As far as I understood, I need to add this lines to my input file:
[Outputs]
  checkpoint = true
[]
 if this is true what should be the command to restart it from the timestep where it stopped after?

Thanks,
Nour

**Reply by GiudGiud** [reply]
That s true

there s a doc page on restart and recover
You ll want to set the restart_file_base in the Problem block or use —recover on the command line with the name of the folder with recover datat

**Comment by nourriahi** [comment]
Hi, 

I have tried to use what you recommended (out_cp file was created), but I think I missed something that I am not aware of somewhere (I followed instructions on the restart and recovery page) which is giving me the error below.
![image](https://github.com/idaholab/moose/assets/131006365/335d5072-527f-4547-a793-73549bfd2ea7)
![image](https://github.com/idaholab/moose/assets/131006365/46edaabe-1eda-4faa-9882-669a84fac7fd)

**Reply by GiudGiud** [reply]
--recover needs a follow up argument with the name of the folder with the checkpoint file

the error message will update to show you what folder it s trying to open

**Reply by nourriahi** [reply]
I tried to do so, but I am still having error messages, I tried with two syntaxes since I am not sure which one should be used:
**_First :_** (with -i after recover)
![image](https://github.com/idaholab/moose/assets/131006365/e58c855b-0255-4c8e-91cb-3ad80ed4c754)
![image](https://github.com/idaholab/moose/assets/131006365/85c8227c-a359-4dd8-b8fe-8907c0e716a9)
**_Second_**(without -i)
![image](https://github.com/idaholab/moose/assets/131006365/d0fe34e0-b413-40f1-91c3-bb3279cd2b1c)
![image](https://github.com/idaholab/moose/assets/131006365/2efd961a-208c-4e04-b1cd-1932460a0243)

**Reply by GiudGiud** [reply]
what does ls return in the directory?
also no need to post screenshot, this is text, it can be copy-pasted, it looks good within triple back-quotes for example

**Reply by nourriahi** [reply]
eddine@mech-lin1:~/projects/additive_manufacturing_3d_convection_radiation_minus/problems$

**Reply by GiudGiud** [reply]
try
`ls ~/projects/additive_manufacturing_3d_convection_radiation_minus/problems`

**Reply by nourriahi** [reply]
additive_manufacturing_3d_convection_radiation_minus_bcs.i
additive_manufacturing_3d_convection_radiation_minus_bcs_out.e
additive_manufacturing_3d_convection_radiation_minus_bcs_packed_100w_subdomain.i
additive_manufacturing_3d_convection_radiation_minus_bcs_packed_100w_z400.i
additive_manufacturing_3d_convection_radiation_minus_bcs_packed_100w_z400_out.e
additive_manufacturing_3d_convection_radiation_minus_bcs_packed_200w_z400.i
additive_manufacturing_3d_convection_radiation_minus_bcs_packed_200w_z400_out.e
additive_manufacturing_3d_convection_radiation_minus_bcs_periods1.i
additive_manufacturing_3d_convection_radiation_minus_bcs_periods1_out.e
additive_manufacturing_3d_convection_radiation_minus_bcs_periods2.i
additive_manufacturing_3d_convection_radiation_minus_bcs_periods2_out.e
additive_manufacturing_3d_convection_radiation_minus_bcs_periods.i
additive_manufacturing_3d_convection_radiation_minus_bcs_periods_out.e
additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_100w_dt.i
additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_100w_dt_out.e
additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_100w_out.e
additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_100w_speed.i
additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_100w_speed_out.e
additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed1.i
additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed1_out.e
additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_200w.i
additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_200w_out.e
additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_200z.i
additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_200z_out.e
additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_300w_400z.i
additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_300w_400z_out_cp
additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_300w_400z_out.e
additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_300w.i
additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_300w_out.e
additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_400w.i
additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_400w_out.e
additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_500w.i
additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_500w_out.e
additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed.i
additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_out.e
data
data_thermal_half
excel100
excel100_400z
excel100speed
excel200
excel200_400z
excel300
excel300_400z
excel400
excel_thermal

**Reply by GiudGiud** [reply]
this is your checkpoint file
`additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_300w_400z_out_cp`

so I would expect
--recover additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_300w_400z_out_cp/some file name here

to work

what does `ls ~/projects/additive_manufacturing_3d_convection_radiation_minus/problems/additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_300w_400z_out_cp` return?

**Reply by nourriahi** [reply]
0015_mesh.cpr  0015.xdr       0016_mesh.cpr  0016.xdr
0015.rd-0      0015.xdr.0000  0016.rd-0      0016.xdr.0000

**Reply by GiudGiud** [reply]
does 
```
--recover additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_300w_400z_out_cp/0016.xdr
```
work?

**Reply by nourriahi** [reply]
when I try from problems or out_cp directories it says recover command not found, and when I tried with : 

> /problems$ ../additive_manufacturing_3d_convection_radiation_minus-opt -i additive_manufacturing_3d_convection_radiation_minus_bcs_packed_100w_z400.i --recover additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_300w_400z_out_cp/0016.xdr

*** Info ***
Using additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_300w_400z_out_cp/0016.xdr for recovery.


*** ERROR ***
Error with file `additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_300w_400z_out_cp/0016.xdr_mesh.cpr'

**Reply by GiudGiud** [reply]
```
--recover additive_manufacturing_3d_convection_radiation_minus_bcs_periods_packed_300w_400z_out_cp/0016
```

**Reply by nourriahi** [reply]
it worked, but, I am not getting the excel data as it was happening previously

**Reply by GiudGiud** [reply]
you mean the csv file?
turn this on/off in the outputs block

**Reply by nourriahi** [reply]
well, it was on in the corresponding input file and csv data were registered in the appropriate folder, but, once I recovered no data are being registered.
```
[Outputs]
  exodus = true
  checkpoint = true
  [csv]
    type = CSV                                                       
    file_base = excel300_400z/additive_manufacturing_temperature_profile_output_300w
    start_time = 0.0
    end_time = 4e-3
    interval = 1
    execute_on = timestep_end
  []
[] 
```

**Reply by GiudGiud** [reply]
Is time still below 4e-3? CSV output stops after that time

**Reply by nourriahi** [reply]
yes it is and even the output file is not changing or increasing in size and last time it was updated is yesterday despite that the simulation is running as it should.

**Reply by GiudGiud** [reply]
try using a different file_base?

**Reply by nourriahi** [reply]
I didnt understand exactly what you meant.

**Reply by GiudGiud** [reply]
file_base is a parameter in the outputs
